auth0/data_source_auth0_client_test.go

```diff
@@ -28,9 +27,7 @@ func TestAccDataClientByName(t *testing.T) {
 	rand := random.String(6)
 
 	resource.Test(t, resource.TestCase{
-		Providers: map[string]*schema.Provider{
```

Comment:
If we ran the test with debug logs enabled we were getting before about 200+ lines of warnings because the provider was double configuring itself due to the usage of the deprecated Providers attribute on the test case. Switching to ProviderFactories instead makes it so it gets configured only once.